### PR TITLE
Added support for create group and add user

### PR DESCRIPTION
**Pull request by samta**

### CODE REVIEW

File: splitwise/__init__.py [hunk 1]
```diff
@@ -261,3 +263,20 @@ def createExpense(self,expense):
...
+    
+    def createGroup(self, name=None, group_type=None, country_code=None):
+        # create expense group
+        group_info = dict(name=name, group_type=group_type, country_code=country_code)
+        content = self.__makeRequest(Splitwise.CREATE_GROUP_URL, "POST", group_info)
+        content = json.loads(content.decode("utf-8"))
+
+        return content
+
+    def addUserToGroup(self, first_name=None, last_name=None, email=None, id=0):
```
Comment (namaggarwal) [inline splitwise/__init__.py]: It should take a user object as input. See createExpense.

id should be group_id for clarity

File: splitwise/__init__.py [hunk 2]
```diff
@@ -261,3 +263,20 @@ def createExpense(self,expense):
             expense = Expense(content["expenses"][0])
 
         return expense
+    
+    def createGroup(self, name=None, group_type=None, country_code=None):
```
Comment (namaggarwal) [inline splitwise/__init__.py]: It should take group object as input, see createExpense

**Review by namaggarwal — CHANGES_REQUESTED** [review]
Please have a look at the comments. I guess while creating a resource we should take resource object as input. This will help us to maintain the code and prevent breaking changes when splitwise adds more field in their APIs

**Comment by samta** [comment]
Thanks for reviewing. I will do the suggested changes and raise a review request again.

One doubt, the create group api page (http://dev.splitwise.com/dokuwiki/doku.php?id=create_group) suggests that we can add members also when try to create group, but when I am posting this api request its not adding members. But the group has got created successfully.

{'members': [{'first_name': '<Name>, 'last_name': '<Title>', 'email': '<email-address>'}], 'country_code': 123, 'name': 'test-group2', 'group_type': 'test-type'}

Can you please suggest ?

**Comment by namaggarwal** [comment]
By looking at the documentation I see that you have to provide json like

{'members': [{'users_1_first_name': ', 'users_1_last_name': '', 'users_1_email': '<email-address>'}], 'country_code': 123, 'name': 'test-group2', 'group_type': 'test-type'}

Please have a look at createExpense function

**Comment by samta** [comment]
Thanks for replying back Naman,

Looks like members is not a valid key, below has worked.
{'users_1_first_name': ', 'users_1_last_name': '', 'users_1_email': '', 'country_code': 123, 'name': 'test-group2', 'group_type': 'test-type'}

**Comment by samta** [comment]
Hi Naman,

Please review.

Here I have taken care of creating group object and let the create group method to take care of calling the api. I have also made a common method to create users as in required format users__0__ ..

I have modified two other files in order to init the object.

If you find the changes fine, I will also checkin example to create group.

File: splitwise/__init__.py [hunk 3]
```diff
@@ -36,8 +36,7 @@ class Splitwise(object):
     GET_EXPENSES_URL    = SPLITWISE_BASE_URL+"api/"+SPLITWISE_VERSION+"/get_expenses"
     GET_EXPENSE_URL     = SPLITWISE_BASE_URL+"api/"+SPLITWISE_VERSION+"/get_expense"
     CREATE_EXPENSE_URL  = SPLITWISE_BASE_URL+"api/"+SPLITWISE_VERSION+"/create_expense"
-    CREATE_GROUP_URL    = SPLITWISE_BASE_URL+"api/"+SPLITWISE_VERSION+"/create_group"
-    ADD_USER_TO_GROUP_URL = SPLITWISE_BASE_URL+"api/"+SPLITWISE_VERSION+"/add_user_to_group"
+    CREATE_GROUP_URL = SPLITWISE_BASE_URL + "api/" + SPLITWISE_VERSION + "/create_group"
```
Comment (namaggarwal) [inline splitwise/__init__.py]: Please align this like other URLs

File: splitwise/__init__.py [hunk 4]
```diff
@@ -263,20 +251,28 @@ def createExpense(self,expense):
...
+            group_members = group.getMembers()
+            del group_info["members"]
+            Splitwise.setUserArray(group_members, group_info)
 
-        content = self.__makeRequest(Splitwise.ADD_USER_TO_GROUP_URL, "POST", user_info)
+        content = self.__makeRequest(Splitwise.CREATE_GROUP_URL, "POST", group_info)
         content = json.loads(content.decode("utf-8"))
-
+        
         return content
```
Comment (namaggarwal) [inline splitwise/__init__.py]: Please return a Group object like in createExpense

File: splitwise/user.py [hunk 5]
```diff
@@ -7,20 +7,26 @@ class User(object):
...
+            else:
+                self.email  = None
+
+            if 'registration_status' in data:
+                self.registration_status = data["registration_status"]
+            else:
+                self.registration_status = None
 
-        self.picture    = Picture(data["picture"])
+            self.picture    = Picture(data["picture"])
```
Comment (namaggarwal) [inline splitwise/user.py]: You can add a condition here if 'picture' in data:

**Review by namaggarwal — CHANGES_REQUESTED** [review]
Hi,

The functionality is fine but the object returned should be a group object. I added some minor comments as well.

Naman

**Pull request merged**